javaslang/src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -409,4 +452,24 @@ default boolean startsWith(Iterable<? extends T> that, int offset) {
             return -1;
         }
     }
+
+    interface Search {
+        static <T> int binarySearch(IndexedSeq<T> seq, Function1<Integer, Integer> comparison) {
```

Comment:
I've searched the API docs and think we can do a little tweak - replacing the `Function1<Integer, Integer>` with `IntUnaryOperator`. In our special case it would prevent auto-boxing/unboxing of int values.